File: datasets/IR-Plag/convertai-original-java-rest-python/case-07/plagiarized/L1/05/multiarray.py
import numpy as np
from typing import List

def sum_major_diagonal(m: List[List[float]]) -> float:
    """
    Calculates the sum of the elements in the major diagonal of a square matrix.
    
    Args:
        m (List[List[float]]): A square matrix represented as a list of lists.
    
    Returns:
        float: The sum of the elements in the major diagonal.
    """
    return sum(m[i][i] for i in range(len(m)))

def main():
    print("Enter a 4 by 4 matrix row by row:")
    m = [list(map(float, input().split())) for _ in range(4)]
    
    print(f"Sum of the elements in the major diagonal is {sum_major_diagonal(m)}")

if __name__ == "__main__":
    main()

File: datasets/IR-Plag/intellij-original-java-rest-kotlin/case-07/plagiarized/L1/05/multiarray.kt
package IR

import java.util.*

object multiarray {
    @JvmStatic
    fun main(args: Array<String>) {
        val input = Scanner(System.`in`)


        print("Enter a 4 by 4 matrix row by row: ")
        val m = Array(4) { DoubleArray(4) } //inisialisai array 2 dimensi

        for (i in 0..3) {
            for (j in 0..3) {
                m[i][j] = input.nextDouble() //menginput angka ke dalam array 2d
            }
        }

        print("Sum of the elements in the major  diagonal is  " + sumMajorDiagonal(m)) //menampilkan fungsi hasil perhitungan diagonal
    }

    fun sumMajorDiagonal(m: Array<DoubleArray>): Double { //fungsi untuk mengitung diagonal
        var sum = 0.0
        for (i in m.indices) {
            sum += m[i][i]
        }
        return sum
    }
}
